### PR TITLE
Add some "basic" ioctl based tests for NVME and AMD-GPU

### PR DESCRIPTION
**Pull request by superm1**
This adds the ability to enumerate and "interact" with IOCTL commands needed to enumerate NVME and AMD-GPU devices.

I think that upgrading NVME will be possible too when we set `CI_NETWORK=1`; but I don't have a device that is supported by LVFS to get all the correct matching metadata to check it.

No documentation yet; this wasn't exactly easy to implement the kernel IOCTL emulation but @benzea's hint was really helpful to get me further along.

CC @martinpitt

Type of pull request:

- [ ] New plugin (Please include [new plugin checklist](https://github.com/fwupd/fwupd/wiki/New-plugin-checklist))
- [x] Code fix
- [ ] Feature
- [ ] Documentation

### CODE REVIEW

**Comment by superm1** [comment]
also thank you very much @vpelletier for the really helpful IOC wrappers ported to Python.

File: plugins/amd-gpu/amd_gpu_test.py [hunk 1]
```diff
@@ -0,0 +1,479 @@
...
+
+
+class drm_amdgpu_info_vbios(ctypes.Structure):
+    _fields_ = [
+        ("name", ctypes.c_char * 64),
+        ("vbios_png", ctypes.c_char * 64),
+        ("version", ctypes.c_uint32),
+        ("pad", ctypes.c_uint32),
+        ("vbios_ver_str", ctypes.c_char * 32),
+        ("date", ctypes.c_char * 32),
```
Comment (hughsie) [inline plugins/amd-gpu/amd_gpu_test.py:32]: it does seem a shame we can't generate these from the .rs files...

File: plugins/nvme/nvme_test.py [hunk 2]
```diff
@@ -0,0 +1,382 @@
...
+A: power/wakeup=disabled
+A: power/wakeup_abort_count=
+A: power/wakeup_active=
+A: power/wakeup_active_count=
+A: power/wakeup_count=
+A: power/wakeup_expire_count=
+A: power/wakeup_last_time_ms=
+A: power/wakeup_max_time_ms=
+A: power/wakeup_total_time_ms=
+A: power_state=D0
```
Comment (hughsie) [inline plugins/nvme/nvme_test.py:309]: do we need all of these? I know it's easier to just dump everything, but for dozens of plugins it's going to be quite verbose.

File: src/fwupd_ioc.py [hunk 3]
```diff
@@ -0,0 +1,146 @@
+# SPDX-License-Identifier: LGPL-2.1+
```
Comment (hughsie) [inline src/fwupd_ioc.py:1]: in src? contrib perhaps?

File: plugins/nvme/nvme_test.py (hunk 2 above)
Comment (superm1) [reply plugins/nvme/nvme_test.py:309]: We don't need everything, I can pare down.

File: src/fwupd_ioc.py (hunk 3 above)
Comment (superm1) [reply src/fwupd_ioc.py:1]: Anything required for build and test in a "normal" build I don't believe should be in contrib/.

File: plugins/amd-gpu/amd_gpu_test.py (hunk 1 above)
Comment (superm1) [reply plugins/amd-gpu/amd_gpu_test.py]: split up

**Comment by superm1** [comment]
I don't want to do it now; but _eventually_ I think it's going to be worth putting all this stuff into a python package instead of all the `PYTHONPATH=` dances that exist right now.  Later we could use it enable use cases _like_ running a totally mocked daemon to test UEFI capsule flows in GNOME-firmware for example.

**Review by hughsie — APPROVED** [review]

**Pull request merged**